test/replication-luatest/gh_7512_raft_leader_hang_test.lua

```diff
@@ -0,0 +1,68 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local fio = require('fio')
+
+local g = t.group('gh-7512')
+
+
+-- gh-7512: Raft followers should notice leader hang and arrange new elections.
+g.before_all(function(cg)
+    cg.cluster = cluster:new{}
+    local box_cfg = {
+        election_mode = 'candidate',
+        replication_timeout = 0.1,
+        replication_synchro_quorum = 1,
+        replication = {
+            server.build_instance_uri('node1'),
+            server.build_instance_uri('node2'),
+        },
+    }
+    cg.node1 = cg.cluster:build_and_add_server{
+        alias = 'node1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.election_mode='voter'
+    cg.node2  = cg.cluster:build_and_add_server{
+        alias = 'node2',
+        box_cfg = box_cfg,
+    }
+    cg.cluster:start()
+
+    cg.unique_filename = server.build_instance_uri('unique_filename')
+    fio.unlink(cg.unique_filename)
+end)
+
+g.after_all(function(cg)
+    cg.cluster:drop()
+    fio.unlink(cg.unique_filename)
+end)
+
+local function block(filename)
+    local fiber = require('fiber')
+    fiber.new(function(filename)
+        -- Can't use fio, because it yields.
+        local cmd = 'sleep 0.1 && ls ' .. filename .. ' > /dev/null 2>&1'
+        while os.execute(cmd) ~= 0 do end
```

Comment:
Would be nice not to depend on having `sleep` command line utility available. FFI `usleep` should help (although I remember last time locally in console I had some problems with it - it was just returning immediately, hmmm). The filename then can be opened using `fio` (it doesn't yield AFAIR). Up to you though, `sleep 0.1` is fine if it works so far.

Reply:
🤷‍♂️ 
```
tarantool> ffi.cdef[[int usleep(int ms);]]
---
...

tarantool> ffi.C.usleep(8900)
---
- 0
...

tarantool> ffi.C.usleep(8901)
---
- -1
...

```

Also I checked, and fio yields, on Mac at least:
```
tarantool> print(fiber.info()[fiber.id()].csw) fio.open('a.out') print(fiber.info()[fiber.id()].csw)
1026
1027
---
...
```
Same results for `fio.listdir`. I guess they all yield.

So I'd rather leave plain `sleep` here.

Reply:
Okay, speaking of `usleep`, it's simply:
```
     The usleep() function will fail if:

     [EINTR]            A signal was delivered to the process and its action was
                        to invoke a signal-catching function.
```
So it must be `SIGURG` setting another fiber slice, I guess?